src/rhsm/utils.py

```diff
@@ -110,24 +108,22 @@ def has_good_scheme(url):
 
 
 def parse_url(
-    local_server_entry,
-    default_hostname=None,
-    default_port=None,
-    default_prefix=None,
-    default_username=None,
-    default_password=None,
-):
+    local_server_entry: str,
+    default_hostname: str = None,
+    default_port: str = None,
+    default_prefix: str = None,
+    default_username: str = None,
+    default_password: str = None,
```

Comment:
It looks like that adding `Union[str, None]` or `Optinal[str]` is redundant, because default value `None` converts it to `Optional[str]` automagically somehow. Try to use this method `parse_url()` somewhere in PyCharm and hint with suggestion will contain `Optional[str]` (not only `str`).